main.go

```diff
@@ -85,20 +87,33 @@ func main() {
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "Warning: failed to determine zone: %s\n", err)
 			zone = ""
+			if *generateMeshId {
+				// Zone is required while using with generateMeshID.
+				os.Exit(1)
+			}
 		}
 	}
 
+	// Generate clusterName from metadata server or from command-line
+	// arguments, with the latter taking preference.
+	cluster := *gkeClusterName
+	if cluster == "" {
+		cluster, err = getClusterName()
+		if err != nil && *generateMeshId {
+			// The metadata server would return an error when run on GCE VMs.
+			// gkeClusterName should be passed in when using generateMeshID in GCE VM
+			// cases.
+			fmt.Fprintf(os.Stderr, "could not discover GKE cluster name: %v", err)
```

Comment:
The other messages are clear whether it is an error or a warning. Follow the similar approach?

Reply:
Done